### PR TITLE
Tested Base Counter

### PR DESCRIPTION
**Pull request by JakeDawkins**
# Feature / Fixed Issue(s)
- Added tests for the counter components and wrappers.
- Changed counter styles to use `@jongold/further`

**Note** @jbaxleyiii the errors we were seeing about perf are from trying to render the recompose wrappers synchronously. `Await`ing those calls fixes this.

# Screenshots
screen shot 2017-04-02 at 10 54 17 am

### CODE REVIEW

**Comment by jbaxleyiii** [comment]
@JakeDawkins I will hug you for this. I love it!

@jongold lots of fun usage patterns for Further coming soon  ❤️

**Comment by newspringweb** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please add your GitHub name ("JakeDawkins") to the changelog entry.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  // create noop function to capture props from withHandlers
+  const noop = props => {
+    increment = props.increment;
+    decrement = props.decrement;
+    return null;
+  };
+
+  beforeEach(async () => {
+    // acutally run actions function
+    const MiniApp = await actions(noop); // await or perf errors
```
Comment (jbaxleyiii) [inline src/__tests__/index.js:67]: This is really interesting to me that it has to be awaited

File: src/index.js [hunk 2]
```diff
@@ -28,26 +29,24 @@ export const actions = withHandlers({
   decrement: ({ setCounter }) => () => setCounter(n => n - 1),
 });
 
+export const counterStyle = Style.of({
```
Comment (jbaxleyiii) [inline src/index.js:32]: 💛

File: src/index.js [hunk 3]
```diff
@@ -28,26 +29,24 @@ export const actions = withHandlers({
...
+
 export const Counter = (
   {
     increment,
     counter,
     decrement,
   }: IWithHandlers,
 ) => (
-  <View style={style.counter}>
+  <View style={counterStyle.resolve()}>
```
Comment (jbaxleyiii) [inline src/index.js:45]: @JakeDawkins I think we should PR adding memoization into further for the resolve function. Same props shouldn't need to recalc the return object each time.

Not for this PR, but just wanted to mention it

**Review by jbaxleyiii — APPROVED** [review]

File: src/index.js (hunk 3 above)
Comment (JakeDawkins) [reply src/index.js:45]: Like this (with ramda)?

screen shot 2017-04-02 at 4 38 15 pm

**Pull request merged**